// @lint-md/core/esm/rules/space-around-number.js
import { markText } from '../utils/mark-text';
const isMarkedTextBetweenChineseAndNumber = (value) => {
    return value === 'ZN' || value === 'NZ';
};
const spaceAroundNumber = {
    meta: {
        name: 'space-around-number'
    },
    create: (context) => {
        return {
            text: (node) => {
                const { value } = node;
                const markedText = markText(value);
                for (let i = 0; i < markedText.length - 1; i++) {
                    const checkStrFragment = markedText.slice(i, i + 2);
                    if (isMarkedTextBetweenChineseAndNumber(checkStrFragment)) {
                        const loc = node.position;
                        context.report({
                            loc: {
                                start: {
                                    line: loc.start.line,
                                    column: loc.start.column + i
                                },
                                end: {
                                    line: loc.start.line,
                                    column: loc.start.column + i + 2
                                }
                            },
                            message: '中文与数字之间需要增加空格',
                            fix: (fixer) => {
                                return fixer.insertTextAt(loc.start.offset + i + 1, ' ');
                            }
                        });
                    }
                }
            }
        };
    }
};
export default spaceAroundNumber;
