// @lint-md/core/lib/rules/correct-title-trailing-punctuation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const get_text_nodes_1 = require("../utils/get-text-nodes");
const FORBIDDEN_PUNCTUATIONS = ['.', ',', ';', ':', '。', '，', '；', '：', '~', '*', '`'];
const correctTitleTrailingPunctuation = {
    meta: {
        name: 'correct-title-trailing-punctuation',
    },
    create: (context) => {
        return {
            heading: (node) => {
                const lastTextNode = (0, get_text_nodes_1.getTextNodes)(node).pop();
                if (lastTextNode) {
                    const val = lastTextNode.value.trimEnd();
                    let endPos;
                    for (endPos = val.length - 1; endPos >= 0; endPos--) {
                        const currentCharacter = val[endPos];
                        if (!FORBIDDEN_PUNCTUATIONS.includes(currentCharacter)) {
                            break;
                        }
                    }
                    if (endPos < val.length - 1) {
                        context.report({
                            loc: node.position,
                            message: '标题末尾不允许出现不规范的标点符号',
                            fix: (fixer) => {
                                return fixer.replaceTextRange([
                                    lastTextNode.position.start.offset,
                                    lastTextNode.position.end.offset
                                ], val.slice(0, endPos + 1));
                            }
                        });
                    }
                }
            }
        };
    }
};
exports.default = correctTitleTrailingPunctuation;
