// @lint-md/core/lib/rules/no-empty-url.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const parser_1 = require("@lint-md/parser");
const handleFixLinkNode = (context, node) => {
    if (node.url.trim() === '') {
        node.url = 'https://example.com';
        let newContent = (0, parser_1.revertMdAstNode)(node);
        if (newContent.endsWith('\n')) {
            newContent = newContent.slice(0, -1);
        }
        context.report({
            loc: node.position,
            message: '链接和图片地址不能为空',
            fix: (fixer) => {
                return fixer.replaceTextRange([
                    node.position.start.offset,
                    node.position.end.offset
                ], newContent);
            }
        });
    }
};
const noEmptyURL = {
    meta: {
        name: 'no-empty-url'
    },
    create: (context) => {
        return {
            link: (node) => {
                handleFixLinkNode(context, node);
            },
            image: (node) => {
                handleFixLinkNode(context, node);
            }
        };
    }
};
exports.default = noEmptyURL;
