// @lint-md/core/lib/rules/no-long-code.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const noLongCode = {
    meta: {
        name: 'no-long-code'
    },
    create: (context) => {
        return {
            code: (node) => {
                const { length: maxLength, exclude = [] } = context.options;
                if (exclude.includes(node.lang)) {
                    return;
                }
                const codeArray = node.value.split('\n');
                for (let i = 0; i < codeArray.length; i++) {
                    if (codeArray[i].length > maxLength) {
                        const firstLineInCodeArea = node.position.start.line;
                        const start = {
                            line: firstLineInCodeArea + i,
                            column: 1
                        };
                        const end = {
                            line: firstLineInCodeArea + i,
                            column: codeArray[i].length
                        };
                        context.report({
                            loc: {
                                start,
                                end
                            },
                            message: '代码块不能有过长的代码'
                        });
                    }
                }
            }
        };
    }
};
exports.default = noLongCode;
