// @lint-md/core/lib/rules/space-around-alphabet.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const mark_text_1 = require("../utils/mark-text");
const isMarkedTextBetweenChineseAndEnglish = (value) => {
    return value === 'ZA' || value === 'AZ';
};
const spaceAroundAlphabet = {
    meta: {
        name: 'space-around-alphabet'
    },
    create: (context) => {
        return {
            text: (node) => {
                const { value } = node;
                const markedText = (0, mark_text_1.markText)(value);
                const boundaries = [];
                for (let i = 0; i < markedText.length - 1; i++) {
                    const checkStrFragment = markedText.slice(i, i + 2);
                    if (isMarkedTextBetweenChineseAndEnglish(checkStrFragment)) {
                        boundaries.push(i);
                    }
                }
                if (boundaries.length > 0) {
                    let pos = 0;
                    let newContent = boundaries.reduce((str, boundary) => {
                        const newContent = `${str}${value.slice(pos, boundary + 1)} `;
                        pos = boundary + 1;
                        return newContent;
                    }, '');
                    newContent += value.slice(pos);
                    context.report({
                        loc: node.position,
                        message: '中英文之间需要添加空格',
                        fix: (fixer) => {
                            return fixer.replaceTextRange([
                                node.position.start.offset,
                                node.position.end.offset
                            ], newContent);
                        }
                    });
                }
            }
        };
    }
};
exports.default = spaceAroundAlphabet;
